public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -501,21 +502,60 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     return [];
   }
 
+  /**
+   * Gets series values
+   * Function to replace old getSeries calls in a way that will provide faster endpoints for new prometheus instances,
+   * while maintaining backward compatability
+   * @param labelName
+   * @param selector
+   */
+  getSeriesValues = async (labelName: string, selector: string): Promise<string[]> => {
+    if (!this.datasource.hasLabelsMatchAPISupport()) {
+      const data = await this.getSeries(selector);
+      return data[labelName] ?? [];
+    }
+    return await this.fetchSeriesValuesWithMatch(labelName, selector);
+  };
+
   /**
    * Fetches all values for a label, with optional match[]
    * @param name
    * @param match
    */
-  fetchSeriesValues = async (name: string, match?: string): Promise<string[]> => {
+  fetchSeriesValuesWithMatch = async (name: string, match?: string): Promise<string[]> => {
     const interpolatedName = name ? this.datasource.interpolateString(name) : null;
     const range = this.datasource.getTimeRangeParams();
     const urlParams = {
       ...range,
-      ...(interpolatedName && { 'match[]': match }),
+      ...(match && { 'match[]': match }),
     };
     return await this.request(`/api/v1/label/${interpolatedName}/values`, [], urlParams);
   };
 
+  /**
+   * Gets series labels
```

Comment:
Provide more context as to why there are two functions needed to replace a single one